python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -654,3 +688,140 @@ def _gather(
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        1    False
+        2    False
+        3     True
+        4     True
+        5    False
+        dtype: bool
+
+        Note
+        ----
+        input Series x and y will not be index aligned, but computed as
+        sequential arrays.
+
+        Note
+        ----
+        poly_ring_offsets must contain only the rings that make up the polygons
+        indexed by poly_offsets. If there are rings in poly_ring_offsets that
+        are not part of the polygons in poly_offsets, results are likely to be
+        incorrect and behavior is undefined.
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether each point falls
+            within the corresponding polygon in the input.
+        """
+        if contains_only_polygons(self) is False:
+            raise TypeError("left series contains non-polygons.")
+
+        (lhs, rhs) = self.align(other) if align else (self, other)
+
+        # RHS conditioning:
+        mode = "POINTS"
+        # point in polygon
+        if contains_only_linestrings(rhs) is True:
+            # condition for linestrings
+            mode = "LINESTRINGS"
+            xy = rhs.lines
+        elif contains_only_polygons(rhs) is True:
+            # polygon in polygon
+            mode = "POLYGONS"
+            xy = rhs.polygons
+        elif contains_only_multipoints(rhs) is True:
+            # mpoint in polygon
+            mode = "MULTIPOINTS"
+            xy = rhs.multipoints
+        else:
+            # no conditioning is required
+            xy = rhs.points
+        xy_points = xy.xy
+        point_indices = xy.point_indices()
+        points = GeoSeries(GeoColumn._from_points_xy(xy_points._column)).points
+
+        # call pip on the three subtypes on the right:
+        point_result = contains_properly(
+            points.x,
+            points.y,
+            lhs.polygons.part_offset[:-1],
+            lhs.polygons.ring_offset[:-1],
+            lhs.polygons.x,
+            lhs.polygons.y,
+        )
+        if (
+            mode == "LINESTRINGS"
+            or mode == "POLYGONS"
+            or mode == "MULTIPOINTS"
+        ):
+            # process for completed linestrings, polygons, and multipoints.
+            # Not necessary for points.
+            result = cudf.DataFrame(
+                {"idx": point_indices, "pip": point_result}
+            )
+            df_result = (
+                result.groupby("idx").sum().sort_index()
```

Comment:
Why do we need to `sort_index` here?

Reply:
Because `groupby` results don't guarantee ordering, I think. I was getting sporadically failing tests wherein the results of the `sum` and `count` groupbys produced different indexes, which then could not be compared via `==`.

Reply:
I do think the last `.sort_index` is no longer required since the comparison is using two index sorted dataframes.